### PR TITLE
Feature/service definition from deployment manifest

### PR DESCRIPTION
**Pull request by melanielc**
Pull request that addresses issue #152

### CODE REVIEW

File: broker/src/functional-test/groovy/com/swisscom/cloud/sb/broker/functional/ServiceDefinitionFromServiceManifestFunctionalSpec.groovy [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+        assert (xxlargePlan.active)
+    }
+
+    /* redis service definition in resources for integration tests service-data/redisServiceToBeUpdated is identical to service definition
+     * in application.yml except for bindable = false. The service definition is created in the DB and then updated using the definition
+     * in the application.yml
+     */
+    def "update service definition by updating it in the service manifest"() {
+        given:
+        serviceBrokerClientExtended = new ServiceBrokerClientExtended(
```
Comment (seilc1) [inline broker/src/functional-test/groovy/com/swisscom/cloud/sb/broker/functional/ServiceDefinitionFromServiceManifestFunctionalSpec.groovy]: could/should be in a separate function e.g. createServiceBrokerClient

File: broker/src/integration-test/groovy/com/swisscom/cloud/sb/broker/servicedefinition/ServiceDefinitionInitializerSpec.groovy [hunk 2]
```diff
@@ -86,20 +73,163 @@ class ServiceDefinitionInitializerSpec extends BaseTransactionalSpecification {
...
+
+        and:
+        ServiceInstance serviceInstance = new ServiceInstance(guid: "testServiceInstance", plan: plan)
+        serviceInstanceRepository.saveAndFlush(serviceInstance)
+
+        when:
+        serviceDefinitionInitializer.init()
+
+        then:
+        assert (!cfServiceRepository.findByGuid(serviceGuid).active)
```
Comment (seilc1) [inline broker/src/integration-test/groovy/com/swisscom/cloud/sb/broker/servicedefinition/ServiceDefinitionInitializerSpec.groovy]: check first that the service is not null ... not needed if everything works fine, but could be nice, when the service is incorrectly deleted

File: broker/src/main/groovy/com/swisscom/cloud/sb/broker/servicedefinition/ServiceDefinitionInitializer.groovy [hunk 3]
```diff
@@ -31,40 +33,88 @@ class ServiceDefinitionInitializer {
...
+            cfServiceMap.put(cfService.guid, cfService)
+        }
+        return cfServiceMap
+    }
 
-        def guidList = cfServiceList.collect {it.guid}
+    void synchroniseServiceDefinitions(List<ServiceDto> services, HashMap<String, CFService> toBeDeleted) {
+        services.each{ service ->
+            addOrUpdateServiceDefinitions(service)
+            toBeDeleted.remove(service.guid)
```
Comment (seilc1) [inline broker/src/main/groovy/com/swisscom/cloud/sb/broker/servicedefinition/ServiceDefinitionInitializer.groovy]: isn't it a Problem if a new service has been added and it is removed from the toBeDeleted List even tough it never was there?

File: broker/src/main/groovy/com/swisscom/cloud/sb/broker/servicedefinition/ServiceDefinitionInitializer.groovy [hunk 4]
```diff
@@ -31,40 +33,88 @@ class ServiceDefinitionInitializer {
...
-            if (!configGuidList.containsAll(guidList)) {
-                throw new RuntimeException("Missing service definition configuration exception. Service list - ${guidList}")
+        toBeDeleted.each{ key, service ->
+            def canDeleteService = true
+            service.plans.toList().each{ plan ->
+                canDeleteService = canDeleteService & tryDeletePlan(plan)
+            }
+            if(canDeleteService) {
+                deleteServiceHibernateCacheSavely(service)
+                cfServiceRepository.flush()
```
Comment (seilc1) [inline broker/src/main/groovy/com/swisscom/cloud/sb/broker/servicedefinition/ServiceDefinitionInitializer.groovy]: should/could this be moved to "deleteServiceHibernateCacheSavely"?

Comment (melanielc) [reply broker/src/main/groovy/com/swisscom/cloud/sb/broker/servicedefinition/ServiceDefinitionInitializer.groovy]: Oh you're totally right, must have overlooked it

File: broker/src/integration-test/groovy/com/swisscom/cloud/sb/broker/servicedefinition/ServiceDefinitionInitializerSpec.groovy (hunk 2 above)
Comment (melanielc) [reply broker/src/integration-test/groovy/com/swisscom/cloud/sb/broker/servicedefinition/ServiceDefinitionInitializerSpec.groovy]: good point

File: broker/src/functional-test/groovy/com/swisscom/cloud/sb/broker/functional/ServiceDefinitionFromServiceManifestFunctionalSpec.groovy (hunk 1 above)
Comment (melanielc) [reply broker/src/functional-test/groovy/com/swisscom/cloud/sb/broker/functional/ServiceDefinitionFromServiceManifestFunctionalSpec.groovy]: agree

File: broker/src/main/groovy/com/swisscom/cloud/sb/broker/servicedefinition/ServiceDefinitionInitializer.groovy (hunk 3 above)
Comment (melanielc) [reply broker/src/main/groovy/com/swisscom/cloud/sb/broker/servicedefinition/ServiceDefinitionInitializer.groovy]: Yes, I've added a check so it's deleted only if is contained in toBeDeleted, cheers

**Pull request merged**